dbms/src/Columns/IColumn.h

```diff
@@ -130,18 +130,29 @@ class IColumn : public COWPtr<IColumn>
 
     /// Appends n-th element from other column with the same type.
     /// Is used in merge-sort and merges. It could be implemented in inherited classes more optimally than default implementation.
+    /// Note: the source column and the destination column must be of the same type, can not ColumnXXX->insertFrom(ConstColumnXXX, ...)
     virtual void insertFrom(const IColumn & src, size_t n) { insert(src[n]); }
 
-    /// Appends range of elements from other column.
+    /// Appends range of elements from other column with the same type.
     /// Could be used to concatenate columns.
+    /// Note: the source column and the destination column must be of the same type, can not ColumnXXX->insertRangeFrom(ConstColumnXXX, ...)
     virtual void insertRangeFrom(const IColumn & src, size_t start, size_t length) = 0;
 
     /// Appends one element from other column with the same type multiple times.
+    /// Note: the source column and the destination column must be of the same type, can not ColumnXXX->insertManyFrom(ConstColumnXXX, ...)
     virtual void insertManyFrom(const IColumn & src, size_t position, size_t length) = 0;
 
     /// Appends disjunctive elements from other column with the same type.
+    /// Note: the source column and the destination column must be of the same type, can not ColumnXXX->insertDisjunctFrom(ConstColumnXXX, ...)
     virtual void insertDisjunctFrom(const IColumn & src, const std::vector<size_t> & position_vec) = 0;
 
+    /// Appends one field multiple times. Can be optimized in inherited classes.
+    virtual void insertMany(const Field & field, size_t length)
```

Comment:
May reduce virtual function call like this:
https://github.com/pingcap/tiflash/blob/adc57e22149d04d76d26f5193134a05a938af7c7/dbms/src/Columns/IColumn.h#L476-L477

Reply:
It should be called as `IColumn->insertMany(v)`

Reply:
Yeah, the insert virutal function is called inside the loop

Reply:
compiler can help optimize this, no vtable lookup here https://gcc.godbolt.org/z/Y48rMoqqc

Reply:
I don't think so, think like this: different derived instances can call insertMany function, thus optimizer can't infer which insert function to be called next. Thus optimizer should lookup the vtable of current instance to find the actual insert function address. That's what "call    qword ptr [rax + 8]" do, I guess.

Reply:
Anyway, I think it is ok to improve it later.